tests/rest/client/test_auth.py

```diff
@@ -1165,3 +1166,371 @@ def _txn(txn: LoggingTransaction) -> int:
+            "/login",
+            content={"type": "m.login.token", "token": login_token},
+        )
+        # It should have failed
+        self.assertEqual(channel.code, 403)
+
+    @override_config(
+        {
+            "oidc_providers": [
+                {
+                    "idp_id": "oidc",
+                    "idp_name": "OIDC",
+                    "issuer": TEST_OIDC_ISSUER,
+                    "client_id": "test-client-id",
+                    "client_secret": "test-client-secret",
+                    "scopes": ["openid"],
+                    "user_mapping_provider": {"config": {}},
+                    "backchannel_logout_enabled": True,
+                }
+            ]
+        }
+    )
+    def test_logout_during_mapping(self) -> None:
+        """
+        It should stop ongoing user mapping session when receiving a logout token
+        """
+        fake_oidc_server = self.helper.fake_oidc_server()
+        user = "john"
+
+        # Get an authentication, and logout before submitting the logout token
+        client_redirect_url = "https://x"
+        userinfo = {"sub": user}
+        channel, grant = self.helper.auth_via_oidc(
+            fake_oidc_server,
+            userinfo,
+            client_redirect_url,
+            with_sid=True,
+        )
+
+        # Expect a user mapping page
+        self.assertEqual(channel.code, HTTPStatus.FOUND, channel.result)
+
+        # We should have a user_mapping_session cookie
+        cookie_headers = channel.headers.getRawHeaders("Set-Cookie")
+        assert cookie_headers
+        cookies: Dict[str, str] = {}
+        for h in cookie_headers:
+            key, value = h.split(";")[0].split("=", maxsplit=1)
+            cookies[key] = value
+
+        user_mapping_session_id = cookies["username_mapping_session"]
+
+        # Getting that session should not raise
+        session = self.hs.get_sso_handler().get_mapping_session(user_mapping_session_id)
+        self.assertIsNotNone(session)
+
+        # Submit a logout
+        logout_token = fake_oidc_server.generate_logout_token(grant)
+        channel = self.submit_logout_token(logout_token)
+        self.assertEqual(channel.code, 200)
+
+        # Now it should raise
+        with self.assertRaises(SynapseError):
+            self.hs.get_sso_handler().get_mapping_session(user_mapping_session_id)
+
+    @override_config(
+        {
+            "oidc_providers": [
+                {
+                    "idp_id": "oidc",
+                    "idp_name": "OIDC",
+                    "issuer": TEST_OIDC_ISSUER,
+                    "client_id": "test-client-id",
+                    "client_secret": "test-client-secret",
+                    "scopes": ["openid"],
+                    "user_mapping_provider": {
+                        "config": {"localpart_template": "{{ user.sub }}"}
+                    },
+                    "backchannel_logout_enabled": False,
+                }
+            ]
+        }
+    )
```

Comment:
Most of these tests seem to have the same config, can we put it in `default_config`?